ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Node/Serialisation.hs

```diff
@@ -6,6 +6,7 @@
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE StandaloneDeriving    #-}
 {-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE UndecidableInstances  #-}
```

Comment:
I'm not anticipating why we'd need to add UndecidableInstances; new `BlockQuery` and old `Query` are the same thing. Is it really needed here?

Same question applies in `ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Serialisation/SerialiseNodeToClient.hs`